A36-xds-for-servers.md

```diff
@@ -237,22 +185,79 @@ prevents replacing the Listener when nothing changes. The grace period should be
+> handling can be skipped.
+
+For each `FilterChain`, if `HttpConnectionManager.route_config` is set, then
+its value should be used, otherwise if the `HttpConnectionManager.rds` field is
+set, an RDS request should be made for the corresponding `RouteConfiguration`
+resource. RouteConfiguration validation is described in the
+[Resource Validation section](#resource-validation).
+
+The server should enter the serving state only after every RDS watcher has
+received some kind of notification, including errors.
+
+#### Configuration Error Logging
+
+There are situations when an XdsServer can clearly tell the configuration will
+cause errors, yet it still applies the configuration. In these situations the
+XdsServer should log a warning each time it receives updates for configuration
+in this state. This is known as "configuration error logging." If an XdsServer
+logs such a warning, then it should also log a single warning once there are no
+longer any such errors. Configuration error logging is currently limited to
+broken RDS resources and an unsupported Route `action` (i.e., is not
+`non_forwarding_action`), both of which cause RPCs to fail with UNAVAILABLE as
+described above.
+
 [Filter.typed_config]: https://github.com/envoyproxy/envoy/blob/928a62b7a12c4d87ce215a7c4ebd376f69c2e080/api/envoy/config/listener/v3/listener_components.proto#L40
 [TypedStruct.type_url]: https://github.com/cncf/udpa/blob/cc1b757b3eddccaaaf0743cbb107742bb7e3ee4f/udpa/type/v1/typed_struct.proto#L38
 [A28-validation]: A28-xds-traffic-splitting-and-routing.md#response-validation
 [Route.non_forwarding_action]: https://github.com/envoyproxy/envoy/blob/5963beae8842982803af1bef04fb5a2a0893c613/api/envoy/config/route/v3/route_components.proto#L242
 [envoy lds]: https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/lds
 
+### Connection and Request Handling
+
+When an XdsServer receives an incoming connection, it should do the following:
+
+ 1. If the state of the server is "not serving", close the connection.
+ 2. Find the most-specifically-matching `FilterChain` for the connection, as
+    described in [the FilterChainMatch section](#filterchainmatch). If nothing
+    matches, use `Listener.default_filter_chain` instead, or if there is no
+    `default_filter_chain`, close the connection.
+ 3. If TLS is supported (see [gRFC A29: xDS TLS Security][A29]), apply the
+    corresponding TLS configuration from the `FilterChain` to the connection.
+
+When the server receives a request on an active connection, it should do the
+following (implementations may reorder these operations):
+
+ 1. If `RouteConfiguration` is supported and if `HttpConnectionManager.rds`
+    references a NACKed resource without a previous good version, an
+    unavailable resource because of communication failures with control plane
+    or a triggered loading timeout, or a non-existent resource, fail the RPC
+    with the UNAVAILABLE status.
```

Comment:
Do we want to say something about what information should be in the status message here?  IIRC, we discussed not making this too specific for security reasons.

Same thing for the next bullet.

Reply:
I added something.